Docker/README.md

```diff
@@ -1,13 +1,86 @@
-# Swift gRPC Base Image
+# Swift gRPC in Docker
 
-Use the Dockerfile in this directory to build a Docker image 
-that's preloaded with the Swift gRPC plugin and related tools.
+Swift gRPC works great within Docker. Use the Dockerfile in this directory to build a Docker image that's preloaded with the Swift gRPC plugin and related tools.
 
-Build with the following command:
+## Prerequisites
 
-    docker build -t grpc/swift .
+- [Docker](https://www.docker.com)
+  - [Docker for Mac](https://hub.docker.com/editions/community/docker-ce-desktop-mac)
+  - [Docker for Windows](https://hub.docker.com/editions/community/docker-ce-desktop-windows)
+  - [Docker for Ubuntu](https://hub.docker.com/editions/community/docker-ce-server-ubuntu)
+  - [Docker for Debian](https://hub.docker.com/editions/community/docker-ce-server-debian)
+  - [Docker for CentOS](https://hub.docker.com/editions/community/docker-ce-server-centos)
+  - [Docker for Fedora](https://hub.docker.com/editions/community/docker-ce-server-fedora)
+
+## Development environment
+
+Build docker image with the following command:
+
+```bash
+docker build -t grpc/swift --target development .
+```
+
+Run docker container with the following command:
+
+```bash
+docker run -it --privileged=true grpc/swift /bin/bash
+```
+
+Run the grpc-swift unit tests:
+
+```bash
+cd grpc-swift
+make test
+```
+
+Run the test client and server:
+
+```bash
+# start the server
+.build/debug/Echo serve &
+# run the client to test each Echo API
+.build/debug/Echo get
+.build/debug/Echo expand
+.build/debug/Echo collect
+.build/debug/Echo update
+#stop the server
+kill -9 `pgrep Echo`
+```
+
+## Protoc-runner
+
+Build docker image with the following command:
+
+```bash
+docker build -t protoc-runner .
+```
+
+If you plan only to use `protoc` with plugins - run `docker image prune` to remove intermediate images.
```

Comment:
I'd rather we kept this section as how to build the image and explaining what it provides instead of how to use Docker.